pkg/v1/sdk/features/plugin/runner.go

```diff
@@ -0,0 +1,196 @@
+	featureGateClient client.Client
+}
+
+// newFeatureGateRunner returns an instance of featureGateRunner.
+func newFeatureGateRunner() (*featureGateRunner, error) {
+	featureGateClient, err := getFeatureGateClient()
+	if err != nil {
+		return nil, err
+	}
+	return &featureGateRunner{featureGateClient: featureGateClient}, nil
+}
+
+// getFeatureGateClient returns a new FeatureGate client
+func getFeatureGateClient() (client.Client, error) {
+	var restConfig *rest.Config
+	var err error
+
+	scheme := runtime.NewScheme()
+	if err := configv1alpha1.AddToScheme(scheme); err != nil {
+		return nil, err
+	}
+	if err := corev1.AddToScheme(scheme); err != nil {
+		return nil, err
+	}
+
+	if kubeConfig == "" {
+		if restConfig, err = k8sconfig.GetConfig(); err != nil {
+			return nil, err
+		}
+	} else {
+		config, err := clientcmd.LoadFromFile(kubeConfig)
+		if err != nil {
+			return nil, fmt.Errorf("failed to load from %s: %w", kubeConfig, err)
+		}
+		rawConfig, err := clientcmd.Write(*config)
+		if err != nil {
+			return nil, fmt.Errorf("unable to write config: %w", err)
+		}
+		if restConfig, err = clientcmd.RESTConfigFromKubeConfig(rawConfig); err != nil {
+			return nil, fmt.Errorf("unable to set up rest config: %w", err)
+		}
+	}
+	mapper, err := apiutil.NewDynamicRESTMapper(restConfig, apiutil.WithLazyDiscovery)
+	if err != nil {
+		return nil, fmt.Errorf("unable to set up rest mapper: %w", err)
+	}
+	crClient, err := client.New(restConfig, client.Options{Scheme: scheme, Mapper: mapper})
+	if err != nil {
+		return nil, fmt.Errorf("unable to create cluster client: %w", err)
+	}
+	return crClient, nil
+}
+
+// getFeatureGate fetches the FeatureGate
+func (f *featureGateRunner) getFeatureGate(ctx context.Context, featureGateName string) (*configv1alpha1.FeatureGate, error) {
+	gate := &configv1alpha1.FeatureGate{}
+
+	if err := f.featureGateClient.Get(ctx, client.ObjectKey{
+		Name: featureGateName,
+	}, gate); err != nil {
+		return nil, fmt.Errorf("couldn't get featuregate %s: %w", featureGateName, err)
+	}
+	return gate, nil
+}
+
+// joinFeatures retrieves additional information of a Feature and merges it with existing information.
+func (f *featureGateRunner) joinFeatures(ctx context.Context, features []*featureInfo) error {
+	for _, info := range features {
+		feature, err := f.getFeature(ctx, info.name)
+		if err != nil {
+			// skip returning the error for unavailable feature when not found
+			// to fetch others
+			if !info.available && strings.Contains(err.Error(), "not found") {
+				continue
+			}
+			return err
+		}
+		info.maturity = feature.Spec.Maturity
+		info.description = feature.Spec.Description
+		info.immutable = feature.Spec.Immutable
+	}
+	return nil
+}
+
+// getFeature fetches the Feature
+func (f *featureGateRunner) getFeature(ctx context.Context, featureName string) (*configv1alpha1.Feature, error) {
+	feature := &configv1alpha1.Feature{}
+
+	if err := f.featureGateClient.Get(ctx, client.ObjectKey{
```

Comment:
You can just `return f.featureGateClient.Get()` and let the caller inspect actual error from apiserver. This will make suggestion in previous comment work (without having to use `errors.Is` function).